lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1466,61 +1463,82 @@ computeAutomaticEnumValueKind(TypeChecker &TC, EnumDecl *ED) {
   }
 }
 
-static void checkEnumRawValues(TypeChecker &TC, EnumDecl *ED) {
+llvm::Expected<bool>
+EnumRawValuesRequest::evaluate(Evaluator &eval, EnumDecl *ED,
+                               TypeResolutionStage stage) const {
   Type rawTy = ED->getRawType();
   if (!rawTy) {
-    return;
+    return true;
   }
-
+  
   if (ED->getGenericEnvironmentOfContext() != nullptr)
     rawTy = ED->mapTypeIntoContext(rawTy);
   if (rawTy->hasError())
-    return;
-
-  // If we don't have a value kind, the decl checker will provide a diagnostic.
-  auto valueKind = computeAutomaticEnumValueKind(TC, ED);
-  if (!valueKind.hasValue())
-    return;
-  
+    return true;
+    
   // Check the raw values of the cases.
   LiteralExpr *prevValue = nullptr;
   EnumElementDecl *lastExplicitValueElt = nullptr;
 
   // Keep a map we can use to check for duplicate case values.
   llvm::SmallDenseMap<RawValueKey, RawValueSource, 8> uniqueRawValues;
 
+  auto uncheckedRawValueOf = [](EnumElementDecl *EED) -> LiteralExpr * {
+    return EED->RawValueExpr;
```

Comment:
Do you really need this lambda?

Reply:
I wanted to make the raw member accesses explicit.

Reply:
Maybe worth a comment, then?